src/theory/splitter.h

```diff
@@ -0,0 +1,123 @@
+/******************************************************************************
```

Comment:
Is "splitter" the best name for this module?  Should it rather be `partition_tracker` or something?

Reply:
What about partitioner? partition_maker? Tracker doesn't really capture that it makes the partitions.

Reply:
`partition_generator`?